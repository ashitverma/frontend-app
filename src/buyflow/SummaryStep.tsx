import React from "react";
import { Link } from "react-router-dom";

interface SummaryStepProps {
  collectedData: {
    firstName: string;
    lastName: string;
    birthDate: Date;
    email: string;
    age: number;
  };
}

const SummaryStep: React.FC<SummaryStepProps> = (props) => {
  return (
    <div data-testid="summary">
      <div>First Name: {props.collectedData.firstName}</div>
      <div>Last Name: {props.collectedData.lastName}</div>
      <div>Birth Date: {props.collectedData.birthDate}</div>
      <div>Email: {props.collectedData.email}</div>
      <div>Age: {props.collectedData.age}</div>
      <div>
        <Link to="/purchased=dev_ins">Purchase</Link>
      </div>
    </div>
  );
};

export default SummaryStep;
