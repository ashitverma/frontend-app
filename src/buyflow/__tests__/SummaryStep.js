import { render } from "@testing-library/react"; // (or /dom, /vue, ...)
import SummaryStep from "../SummaryStep";
import { MemoryRouter } from "react-router-dom";

it("renders a summary of all the details entered", async () => {
  const collectedDataMock = {
    firstName: "john",
    lastName: "doe",
    birthDate: "1998-09-19",
    email: "johndoe@test.com",
    age: "43",
  };
  const queries = render(<SummaryStep collectedData={collectedDataMock} />, {
    wrapper: MemoryRouter,
  });
  expect(queries.getByText(/purchase/i)).toBeInTheDocument();
});

it("should take a snapshot", () => {
  const collectedDataMock = {
    firstName: "john",
    lastName: "doe",
    birthDate: "1998-09-19",
    email: "johndoe@test.com",
    age: "43",
  };
  const { asFragment } = render(
    <SummaryStep collectedData={collectedDataMock} />,
    {
      wrapper: MemoryRouter,
    }
  );

  expect(asFragment(<SummaryStep />)).toMatchSnapshot();
});
