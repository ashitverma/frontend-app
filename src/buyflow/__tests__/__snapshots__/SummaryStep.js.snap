// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="summary"
  >
    <div>
      First Name: john
    </div>
    <div>
      Last Name: doe
    </div>
    <div>
      Birth Date: 1998-09-19
    </div>
    <div>
      Email: johndoe@test.com
    </div>
    <div>
      Age: 43
    </div>
    <div>
      <a
        href="/purchased=dev_ins"
      >
        Purchase
      </a>
    </div>
  </div>
</DocumentFragment>
`;
